ACPs/194-streaming-asynchronous-execution/README.md

```diff
@@ -281,89 +363,46 @@ $$
-    t2  :t2, after t1, 0.4s
-    t3  :t3, after t2, 0.8s
-    t4  :t4, after t3, 0.9s
-    t5  :t5, after t4, 2s
-    t6  :t6, after t5, 1.1s
-    section Donations
-    c1->c2  : 800, 0.2s
-    c3->c4  : 2900, 0.1s
-    c4->c5  : 2900, 1.1s
-    c5->c6  : 4900, 0.1s
-```
+### JSON RPC methods
 
-1. Total gas used by transactions `t0` and `t1` is less than a single chunk so they are allocated to `c1`.
-2. Addition of `t2` would exceed `c1` capacity so it is instead allocated to `c2`.
-3. Note the necessary "donation" of `c1`'s residual capacity to `c2`, allowing for allocation of `t3` as well.
-4. Transaction `t4` is the only one that can fit into chunk `c3`.
-5. The large transaction `t5` requires more than a single chunk of gas, resulting in `c4` being an empty _interim_ chunk and the allocation of `t5` to `c5`.
-    1. The residual of `c3` was first donated to `c4`, which was unable to fit the next transaction, resulting in transitive donation to `c5`.
-6. Even with its double donation, `c5` no longer has sufficient capacity for the final transaction `t6`, which is in `c6`.
+Although asynchronous execution decouples the transactions and receipts recorded by a specific block, APIs MUST NOT alter their behavior to mirror this.
+In particular, the API method `eth_getBlockReceipts` MUST return the receipts corresponding to the block's transactions, not the receipts settled in the block.
 
-Note that the donation-adjusted chunks fully utilise chain capacity while also completing execution no later than their respective timestamps, which would otherwise be detrimental to consensus.
+#### Named blocks
 
-```mermaid
----
-displayMode: compact
----
-gantt
-    title Donation-adjusted chunks
-    dateFormat x
-    axisFormat %S
-    tickInterval 1second
-    section Chunks
-    c1  :a1, 0, 0.8s
-    c2  :a2, after a1, 1.2s
-    c3  :a3, after a2, 0.9s
-    c5  :a5, after a3, 2s
-    c6  :a6, after a5, 1.1s
-    section Transactions
-    t0  :t0, 0, 0.4s
-    t1  :t1, after t0, 0.4s
-    t2  :t2, after t1, 0.4s
-    t3  :t3, after t2, 0.8s
-    t4  :t4, after t3, 0.9s
-    t5  :t5, after t4, 2s
-    t6  :t6, after t5, 1.1s
-```
+The Ethereum Mainnet APIs allow for retrieving blocks by named parameters that the API server resolves based on their consensus mechanism.
+Other than the _earliest_ (genesis) named block, which MUST be interpreted in the same manner, all other named blocks are mapped to SAE in terms of the execution status of blocks and MUST be interpreted as follows:
+
+ * _pending_: the most recently _accepted_ block;
+ * _latest_: the block that was most recently _executed_;
+ * _safe_ and _finalized_: the block that was most recently _settled_.
+
+> [!NOTE]
+> The finality guarantees of Snowman consensus remove any distinction between _safe_ and _finalized_. 
+> Furthermore, the _latest_ block is not at risk of re-org, only of a negligible risk of data corruption local to the API node.
 
 ### Streaming vs regular asynchronous pipelining
```

Comment:
This diagram is no longer relevant as it applied to the per-second chunking approach.